devicetypes/smartthings/aeotec-doorbell-6.src/aeotec-doorbell-6.groovy

```diff
@@ -0,0 +1,238 @@
+        capability "Alarm"
+
+        fingerprint mfr: "0371", prod: "0003", model: "00A2", deviceJoinName: "Aeotec Doorbell 1"
+    }
+
+    tiles {
+        multiAttributeTile(name: "alarm", type: "generic", width: 6, height: 4) {
+            tileAttribute("device.alarm", key: "PRIMARY_CONTROL") {
+                attributeState "off", label: 'off', action: 'alarm.siren', icon: "st.alarm.alarm.alarm", backgroundColor: "#ffffff"
+                attributeState "both", label: 'ring!', action: 'alarm.off', icon: "st.alarm.alarm.alarm", backgroundColor: "#0e7507"
+            }
+        }
+        standardTile("off", "device.alarm", inactiveLabel: false, decoration: "flat") {
+            state "default", label: '', action: "alarm.off", icon: "st.secondary.off"
+        }
+        standardTile("reset", "device.reset", height: 2, width: 2, inactiveLabel: false, decoration: "flat") {
+            state "default", label: "reset", action: "reset"
+        }
+        valueTile("tamper", "device.tamper", height: 2, width: 2, decoration: "flat") {
+            state "clear", label: 'tamper clear', backgroundColor: "#ffffff"
+            state "detected", label: 'tampered', backgroundColor: "#ffffff"
+        }
+
+        main "alarm"
+        details(["alarm", "off", "tamper"])
+    }
+}
+
+def installed() {
+    initialize()
+    def numberOfSounds = 8
+    state.soundTable = []
+    for (def i in 0..numberOfSounds) {
+        state.soundTable[i] = 0
+    }
+    sendEvent(name: "alarm", value: "off", isStateChange: true)
+}
+
+def updated() {
+    initialize()
+}
+
+def initialize() {
+    if (!childDevices) {
+        addChildren()
+    }
+    sendEvent(name: "checkInterval", value: 2 * 60 * 60 + 2 * 60, displayed: false, data: [protocol: "zwave", hubHardwareId: device.hub.hardwareID, offlinePingable: "1"])
+}
+
+def parse(String description) {
+    def result = null
+    def cmd = zwave.parse(description)
+    if (cmd) {
+        result = zwaveEvent(cmd)
+    }
+    log.debug "Parse returned: ${result.inspect()}"
+    return result
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.basicv1.BasicReport cmd) {
+    createEvents(cmd.value)
+}
+
+def createEvents(value) {
+    sendEvent(name: "alarm", value: value ? "both" : "off")
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.multichannelv3.MultiChannelCmdEncap cmd) {
+    def encapsulatedCommand = cmd.encapsulatedCommand()
+    def endpoint = cmd.sourceEndPoint
+    if (endpoint == 1) {
+        zwaveEvent(encapsulatedCommand)
+    } else {
+        String childDni = "${device.deviceNetworkId}:$endpoint"
+        def child = childDevices.find { it.deviceNetworkId == childDni }
+
+        child?.handleZWave(encapsulatedCommand)
+    }
+}
+
+def on() {
+    setActiveSound(1)
+    def cmds = []
+    cmds <<encap(zwave.basicV1.basicSet(value: 0xFF).format())
+    cmds
+}
+
+def off() {
+    setActiveSound(1)
+    def cmds = []
+    cmds <<encap(zwave.basicV1.basicSet(value: 0x00).format())
```

Comment:
what's the point of calling `encap` on this if you're not passing a second argument?